sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -260,9 +260,9 @@ private object RowToColumnConverter {
       case BooleanType => BooleanConverter
       case ByteType => ByteConverter
       case ShortType => ShortConverter
-      case IntegerType | DateType => IntConverter
+      case IntegerType | DateType | _: YearMonthIntervalType => IntConverter
       case FloatType => FloatConverter
-      case LongType | TimestampType => LongConverter
+      case LongType | TimestampType | _: DayTimeIntervalType => LongConverter
```

Comment:
@gengliangwang shall we handle timestamp NTZ here?

Reply:
Yes